synapse/logging/context.py

```diff
@@ -689,10 +689,10 @@ def run_in_background(f, *args, **kwargs) -> defer.Deferred:
     current = current_context()
     try:
         res = f(*args, **kwargs)
-    except:  # noqa: E722
+    except Exception as e:
```

Comment:
explicit > implicit

I know that `defer.fail()` will take the current exception even if this wasn't captured, but imo it's more friendly to read this way.